### PR TITLE
[test] Add some tests for swift-api-digester's data generation for complex type changes.

### PR DESCRIPTION
**Pull request by nkcsgexi**

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test and merge

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test and merge

File: test/api-digester/Outputs/macro-gen.def [hunk 1]
```diff
@@ -6,6 +6,17 @@ SDK_CHANGE(Function, TypeRewritten, "0", "s:12macrogenleft2C1C4foo2SSSi_SitF", "
...
+SDK_CHANGE(Function, UnwrapOptional, "2", "s:12macrogenleft2S1V4foo7SiSgSi1x_AE1ytF", "", "", "", "macrogenleft")
+SDK_CHANGE(Function, ImplicitOptionalToOptional, "0", "s:12macrogenleft2S1V4foo8SQySiGSi1x_AE1ytF", "", "", "", "macrogenleft")
+SDK_CHANGE(Function, ImplicitOptionalToOptional, "2", "s:12macrogenleft2S1V4foo8SQySiGSi1x_AE1ytF", "", "", "", "macrogenleft")
+SDK_CHANGE(Function, WrapOptional, "2:0", "s:12macrogenleft2S1V4foo9ySays10DictionaryVySiSSSgGG1x_SaySiG1ytF", "", "", "", "macrogenleft")
+SDK_CHANGE(Function, UnwrapOptional, "1:0:1", "s:12macrogenleft2S1V4foo9ySays10DictionaryVySiSSSgGG1x_SaySiG1ytF", "", "", "", "macrogenleft")
+SDK_CHANGE(Function, WrapOptional, "1:1:1:1:1:0", "s:12macrogenleft2S1V5foo10ySSSi_s10DictionaryVySiAFySSSaySiGGGtc1x_tF", "", "", "", "macrogenleft")
+SDK_CHANGE(Function, TypeRewritten, "1:1:1:1", "s:12macrogenleft2S1V5foo11ySSSi_s10DictionaryVySiAFySSSaySiGGGtc1x_tF", "", "[String : [Int]]", "Int", "macrogenleft")
+SDK_CHANGE(Function, TypeRewritten, "1:1:1:1:1:0:0:0:0", "s:12macrogenleft2S1V5foo12ySSSi_s10DictionaryVySiAFySSSaySiGGGtc1x_tF", "", "Int", "String", "macrogenleft")
```
Comment (nathawes) [inline test/api-digester/Outputs/macro-gen.def]: Hey Xi, do the parens count as levels in `ChildIndex`? Was just wondering if this should be `1:1:1:1:1:0`.

Comment (nkcsgexi) [inline test/api-digester/Outputs/macro-gen.def]: it is one level when we compare modules, but the framework authors rarely write type representations like this. I put the case here just for testing purpose. My original concern is that users' code may insert extra parens, we should ignore them as levels.

**Pull request merged**